### DISCUSSION TITLE
An event before onsubmit and handleSubmit

### DISCUSSION THREAD

**Discussion by alvamanu** [react-hook-form/react-hook-form#1669] [General] [answered]
I'm trying to figure out an event between the handleSubmit from useForm and my final onSubmit event. If instead of me doing 
`onClick={handleSubmit(onSubmit)} `
I do
`onClick={handleSubmitAttempt} `
And in handleSubmitAttempt I put handleSubmit(onSubmit) it won't work.

I need to figure out how to put a function between both functions. Thanks for any help.

**Comment by bluebill1049** [comment, marked as answer]
This should worked ->`onClick={() => handleSubmit(onSubmit)()}`

**Comment by alvamanu** [comment]
Thanks buddy. On point.
